src/subscriber_input.cpp

```diff
@@ -11,19 +11,19 @@ SubscriberInput::SubscriberInput(const std::string& name, const NodeConfiguratio
              StatefulActionNode(name, config), node_(node), input_topic_(input_topic)
 {
   RCLCPP_INFO(LOGGER, "SubscriberInput Stateful constructor");
-}
-
-NodeStatus SubscriberInput::onStart()
-{
   callback_group_ = node_->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
   auto options = rclcpp::SubscriptionOptions();
   options.callback_group = callback_group_;
-  RCLCPP_INFO(LOGGER, "Subscribing to Input Topic: %s", input_topic_.c_str());
-  msg_received_ = false;
+
   empty_msg_sub_ = node_->create_subscription<std_msgs::msg::Empty>(
```

Comment:
I would say keeping the subscriber up between calls to tick() really depends on what you are doing. For example if you are subscribing to point clouds in a node which it is infrequently ticked it might not be worth it. 99% of the messages could be thrown away but the trade off would be to take a small hit each time the leaf node is activated to spin up the subscriber.